### PR TITLE
WIP slick.core

### LINKED ISSUE #21: [ts conversion] slick.core.js
Resolved by #42

### PR DESCRIPTION
**Pull request by JacobMGEvans**
Anyone feel free to work off this. While I am not logged in to Discord. Can communicate with me when I am to work with me on it. 

Some of the typing is crude, it is more of a "get it to work" situation

### CODE REVIEW

**Comment by JacobMGEvans** [comment]
I think all the constructor functions will need a constructor signature casted onto them. So far I only added it to ones that caused issues (were being used locally in the file)

File: slick.core.ts [hunk 1]
```diff
@@ -5,13 +5,15 @@
...
+  type TEventData = Event & JQuery.Event;
+
   /***
    * An event object for passing data to event handlers and letting them control propagation.
    * <p>This is pretty much identical to how W3C and jQuery implement events.</p>
    * @class EventData
    * @constructor
    */
-  function EventData() {
+  const EventData = (function (this: TEventData): void {
```
Comment (mpeyper) [inline slick.core.ts:16]: does `function EventData(this: TEventData): void {` not work too?

Comment (JacobMGEvans) [reply slick.core.ts:16]: No I tried that, but if you can get it to work, I would prefer that over the function expression

Comment (JacobMGEvans) [reply slick.core.ts:16]: I need it to be a function expression to recast it with a constructor signature

**Comment by JacobMGEvans** [comment]
Types working for core
- Caveat the types are crude
more generic types and uses types from other modules
would greatly benefit the core module

File: slick.core.ts (hunk 1 above)
Comment (rbusquet) [reply slick.core.ts:16]: could this be an actual class?

Comment (rbusquet) [reply slick.core.ts:16]: I understand the footprint of the change would be much larger

Comment (JacobMGEvans) [reply slick.core.ts:16]: Yeah, that is basically why I didn't use the actual class. 

Fundamentally this is the same and the type system understands what it is.

File: @scripts/slick.remotemodel-yahoo.js [hunk 2]
```diff
@@ -75,7 +75,8 @@
                     url: url,
                     callbackParameter: "callback",
                     cache: true,
-                    success: function (json, textStatus, xOptions) {
+                    success: function (json, _textStatus, _xOptions) {
```
Comment (JacobMGEvans) [inline @scripts/slick.remotemodel-yahoo.js:78]: I am not even certain those two prefixed params are needed

File: plugins/slick.state.ts [hunk 3]
```diff
@@ -123,12 +123,12 @@ type SlickStateFunction = (options: SlickStateOptions) => void;
               var defaultColumns = options.defaultColumns;
               if (defaultColumns) {
                 var defaultColumnsLookup: DefaultColumns = {};
-                $.each(defaultColumns, function (idx, colDef) {
+                $.each(defaultColumns, function (_, colDef) {
```
Comment (JacobMGEvans) [inline plugins/slick.state.ts:126]: underscores get skipped my TS checks for param usage

File: slick.remotemodel-yahoo.ts [hunk 4]
```diff
@@ -94,7 +94,8 @@
           url: url,
           callbackParameter: "callback",
           cache: true,
-          success: function (json, textStatus, xOptions) {
+          success: function (json, _textStatus, _xOptions) {
```
Comment (JacobMGEvans) [inline slick.remotemodel-yahoo.ts]: I am not even certain those two prefixed params are needed

Comment (mpeyper) [reply slick.remotemodel-yahoo.ts]: If they're unused you should be able to drop them.

Comment (JacobMGEvans) [reply slick.remotemodel-yahoo.ts]: Agreed. I will do that.

**Pull request merged**